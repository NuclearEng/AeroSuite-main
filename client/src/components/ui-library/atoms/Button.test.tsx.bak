import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';
import { ThemeProvider } from '@mui/material/styles';
import { createTheme } from '@mui/material/styles';

// Create a test theme to ensure consistent snapshots
const testTheme = createTheme({
  palette: {
    primary: {
      main: '#1976d2',
    },
    secondary: {
      main: '#dc004e',
    },
  },
});

describe('Button Component', () => {
  it('renders correctly with default props', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={testTheme}>
          <Button>Click me</Button>
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when disabled', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={testTheme}>
          <Button disabled>Click me</Button>
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when loading', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={testTheme}>
          <Button isLoading>Click me</Button>
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with custom loading text', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={testTheme}>
          <Button isLoading loadingText="Please wait...">Click me</Button>
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with different variants', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={testTheme}>
          <div>
            <Button variant="contained" color="primary">Contained</Button>
            <Button variant="outlined" color="secondary">Outlined</Button>
            <Button variant="text">Text</Button>
          </div>
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
}); 