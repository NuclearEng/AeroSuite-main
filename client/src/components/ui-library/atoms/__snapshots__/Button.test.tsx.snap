// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly when disabled 1`] = `null`;

exports[`Button Component renders correctly when loading 1`] = `null`;

exports[`Button Component renders correctly with custom loading text 1`] = `null`;

exports[`Button Component renders correctly with default props 1`] = `null`;

exports[`Button Component renders correctly with different variants 1`] = `null`;
